packages/framework/ai-collab/src/explicit-strategy/agentEditReducer.ts

```diff
@@ -172,9 +173,19 @@ export function applyAgentEdit(
 			const node = getNodeFromTarget(treeEdit.target, idGenerator);
 			const { treeNodeSchema } = getSimpleNodeSchema(node);
 
-			const fieldSchema =
-				(treeNodeSchema.info as Record<string, ImplicitFieldSchema>)[treeEdit.field] ??
-				fail("Expected field schema");
+			const nodeFieldSchemas = treeNodeSchema.info as Record<string, ImplicitFieldSchema>;
```

Comment:
How certain are we that this cast is always correct? Wondering if we need some runtime check here.

Reply:
This will only be true for `Object` schema. If we know this will always be an object, then we should probably cast `treeNodeSchema` to its object implementation (ideally with a type guard in place to ensure we catch an unexpected case where it isn't an object). Then `.info` should have the right type without need to cast it.

Reply:
Josh is correct - you would narrow the node schema with `treeNodeSchema instanceof ObjectNodeSchema` or `isObjectNodeSchema(treeNodeSchema)`, and then `info` will be strongly typed.

I think both of those capabilities are currently internal. If that's indeed the case, and you can't use them here, feel free to file a work item for SharedTree team to expose them and put a TODO here in the meantime.

Reply:
Thanks for the feedback. I've updated the code to include a TODO since those methods are not exported by SharedTree.

Reply:
Thank you!